hal/src/main/native/sim/Notifier.cpp

```diff
@@ -157,6 +157,12 @@ uint64_t HAL_WaitForNotifierAlarm(HAL_NotifierHandle notifierHandle,
 
   std::unique_lock lock(notifier->mutex);
   while (notifier->active) {
+    uint64_t curTime = HAL_GetFPGATime(status);
+    if (notifier->running && curTime >= notifier->waitTime) {
+      notifier->running = false;
+      return curTime;
+    }
+
     double waitTime;
     if (!notifier->running || notifiersPaused) {
       waitTime = (HAL_GetFPGATime(status) * 1e-6) + 1000.0;
```

Comment:
This should use curTime instead of calling HAL_GetFPGATime again?

Reply:
Whoops, yes, I meant to do that but missed it.